pkg/sql/catalog/schematelemetry/schema_telemetry_job.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package schematelemetry
+
+import (
+	"context"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/kv"
+	"github.com/cockroachdb/cockroach/pkg/server/telemetry"
+	"github.com/cockroachdb/cockroach/pkg/settings"
+	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
+	"github.com/cockroachdb/cockroach/pkg/sql"
+	"github.com/cockroachdb/cockroach/pkg/sql/sqltelemetry"
+	"github.com/cockroachdb/errors"
+)
+
+var (
+	jobEnabled = settings.RegisterBoolSetting(
+		settings.TenantWritable,
+		"sql.schema_telemetry.job.enabled",
+		"whether the schema telemetry job is enabled",
+		true,
+	).WithPublic()
+)
+
+type schemaTelemetryResumer struct {
+	job *jobs.Job
+	st  *cluster.Settings
+}
+
+var _ jobs.Resumer = (*schemaTelemetryResumer)(nil)
+
+// Resume implements the jobs.Resumer interface.
+func (t schemaTelemetryResumer) Resume(ctx context.Context, execCtx interface{}) error {
+	p := execCtx.(sql.JobExecContext)
+
+	if enabled := jobEnabled.Get(p.ExecCfg().SV()); !enabled {
+		return errors.Newf("schema telemetry jobs are currently disabled by CLUSTER SETTING %s", jobEnabled.Key())
+	}
+
+	telemetry.Inc(sqltelemetry.SchemaTelemetryExecuted)
+
+	var knobs sql.SchemaTelemetryTestingKnobs
+	if k := p.ExecCfg().SchemaTelemetryTestingKnobs; k != nil {
+		knobs = *k
+	}
+
+	aostDuration := -time.Second * 30
```

Comment:
what's the rationale for 30 seconds in the past?

Reply:
It's something I saw in the context of row-level TTLs, I assume it's to reduce the impact of doing full table scans on the namespace and descriptor tables. Should this be a cluster setting instead? I'm not quite sure.

Reply:
There are two benefits to using a past timestamp:
1) Avoid contention
2) Read from followers

For 2) the condition is based on some cluster settings. I think that it's a mistake to have this just be a constant. I'd be okay with it being the further in the past of some constant. See here for how to compute the follower read timestamp:
https://github.com/cockroachdb/cockroach/blob/efb99b36e35866b7a81d8298b65ab39f5b6d6203/pkg/sql/sem/builtins/builtins.go#L9594-L9609